### DISCUSSION TITLE
How would you force close a websocket after a timeout?

### DISCUSSION THREAD

**Discussion by Zanoro9000** [uNetworking/uWebSockets.js#677] [Q&A] [answered]
I have an authentication process for devices and would like that to expire every specified period of time (e.g. 24hrs after they connected). I don't see any way to close a single connection from outside of a message handler. 
I would prefer not to have to put the work of closing on the client side or to have to restart the whole server. I could invalidate their auth status in the socket and then on their next message close them but wondering if there is an easier way.
Note that they auto re-auth whenever they connect/reconnect, and an external source handles the initial authentication.

**Comment by e3dio** [comment, marked as answer]
```javascript

app.ws('/', {
   open: ws => {
      ws.timeout = setTimeout(() => ws.close(), timeout)
   },
   close: ws => {
      clearTimeout(ws.timeout)
   }
})
```

**Reply by Zanoro9000** [reply]
Ah I misread the docs and thought the ws object was only valid during the duration of the request. On reread you are correct that they are valid from open to close. Thanks!
